pkg/config/auto_listener.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package config
+
+import (
+	"os"
+	"path"
+
+	log "github.com/cihub/seelog"
+)
+
+// AutoAddListeners adds the docker listener if needed
+// TODO: support more listeners (kubelet, ...)
+func AutoAddListeners(listeners []Listeners) []Listeners {
```

Comment:
The semantic is not clear to me: what happens if I select "auto" and "docker"? Or, in the future, "auto" and "kubelet"?